### PR TITLE
feat: Don't use Magic numbers

### PR DESCRIPTION
**Pull request by ynsehoornenborg**
This Analyzer checks rule [10@301](https://tics.tiobe.com/viewerCS/?ID=129&CSTD=Rule) of the Philips C# Coding Standard.

Only the following numbers are allowed inline, rest of them needs to be defined as static field or constant.
- 0
- power of 2
- power of 10
- degrees or right angles

### CODE REVIEW

File: Philips.CodeAnalysis.Common/DiagnosticIds.cs [hunk 1]
```diff
@@ -110,5 +110,6 @@ public enum DiagnosticIds
 		AvoidMultipleLambdasOnSingleLine = 2115,
 		AvoidArrayList = 2116,
 		AvoidUnnecessaryWhere = 2117,
+		DontUseMagicNumbers = 2118,
```
Comment (bcollamore) [inline Philips.CodeAnalysis.Common/DiagnosticIds.cs:113]: Just for consistency, rename to AvoidMagicNumbers

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/DontUseMagicNumbersAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+			{
+				current *= bas;
+			}
+
+			return nut == current;
+		}
+
+		private static bool IsStaticOrConst(FieldDeclarationSyntax field)
+		{
+			bool isStatic = field.Modifiers.IndexOf(SyntaxKind.StaticKeyword) >= 0;
```
Comment (bcollamore) [inline Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/DontUseMagicNumbersAnalyzer.cs:100]: Modifiers.Any(SyntaxKind.StaticKeyword)

File: Philips.CodeAnalysis.Test/Maintainability/Maintainability/DontUseMagicNumbersAnalyzerTest.cs [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+		private const string CorrectUnsigned = @"
+namespace DontUseMagicNumbersTests {
+    public class Number {
+        private uint NoMagic = 0;
+    }
+}";
+
+		private const string CorrectLong = @"
+namespace DontUseMagicNumbersTests {
```
Comment (bcollamore) [inline Philips.CodeAnalysis.Test/Maintainability/Maintainability/DontUseMagicNumbersAnalyzerTest.cs:39]: -1?

**Review by bcollamore — DISMISSED** [review]

File: Philips.CodeAnalysis.Test/Maintainability/Maintainability/DontUseMagicNumbersAnalyzerTest.cs (hunk 3 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.Test/Maintainability/Maintainability/DontUseMagicNumbersAnalyzerTest.cs:39]: Sure, let's be gently and add it

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/DontUseMagicNumbersAnalyzer.cs (hunk 2 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/DontUseMagicNumbersAnalyzer.cs]: Replaced

File: Philips.CodeAnalysis.Common/DiagnosticIds.cs (hunk 1 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.Common/DiagnosticIds.cs:113]: Done

**Review by bcollamore — APPROVED** [review]

**Pull request merged**